sherpa/astro/xspec/tests/test_xspec_unit.py

```diff
@@ -1661,3 +1661,256 @@ def test_xspec_model_kwarg_cached_not_needed():
     assert mdl._cache_ctr['misses'] == 3
 
     assert y3 == pytest.approx(y1)
+
+
+@requires_xspec
+@requires_data
+@requires_fits
+@pytest.mark.parametrize("addmodel", [0, 1])
+@pytest.mark.parametrize("redshift", [0, 1])
+@pytest.mark.parametrize("escale", [0, 1])
+def test_table_mod_negative_delta_1850(addmodel, redshift, escale, make_data_path):
+    """Is delta<0 recognized as freezing a parameter?
+
+    The tables in xspec_table_models are numbered in such a way that
+    we can loop through them all, and hence check some other things at
+    the same time (to avoid having to read things in multiple times).
+
+    """
+
+    from sherpa.astro import xspec
+
+    name = f"smod{addmodel}{redshift}{escale}.tmod"
+    infile = make_data_path(f"xspec_table_models/{name}")
+
+    tbl = xspec.read_xstable_model('tbl', infile)
+
+    # issue #1850; is the parameter recognized as frozen
+    assert tbl.lscale.frozen
+
+    parnames = [p.name for p in tbl.pars]
+    assert parnames[0] == "lscale"
+
+    idx = 1
+    if redshift:
+        assert parnames[idx] == "redshift"
+        assert tbl.redshift.frozen
+        idx += 1
+
+    if escale:
+        assert parnames[idx] == "Escale"
+        assert tbl.escale.frozen
+        idx += 1
```

Comment:
Should we also check (here or in a different location) that 
```
if not escale:
    assert 'Escale' not in parnames
```
or something like that?

I don't know the XSPEC table models as well. Of course, such an error should not happen, but since this PR special-cases escale, redshift, addmodel it would probably not hurt, right?

Reply:
So, as mentioned in the main text, we already test this case because we check each element of parnames in this loop:

- parnames[0] is lscale
- parnames[idx] is redshift and or Escale if expected, with idx increasing each time when relevant
- parnames[idx] is norm if an addmodel case, with idx increasing
- len([parnames] == idx